src/components/TarsiaGrid/index.jsx

```diff
@@ -1,33 +1,33 @@
 import React from 'react'
 import Triangle from '../Triangle'
-
 import appConfig from '../../config'
 
-const TarsiaGrid = ({ id, config, values }) => {
-  const grid = []
-  for (let tri of config) {
-    let valueArray = []
-    for (let valueCode of tri.values) { // lookup the value code from the config
-      if (valueCode) { // handles nulls
-        let valueCodeSplit = [valueCode.substr(0,valueCode.length-1), valueCode.slice(-1)] // split into number and q/a
-        if (values[valueCodeSplit[0]]) { // handles nulls
-          valueArray.push(values[valueCodeSplit[0]][valueCodeSplit[1]]) // add the value based on the question number and q/a
-        } else (
-          valueArray.push(valueCode)
-        )
-      } else {
-        valueArray.push(null) // null handling
+const TarsiaGrid = ({ id, grid, values }) => {
+  const formatGridLocationValueArray = ({ values, locationValues }) => locationValues.map(value => {
+    if (value) {
+      // split into number and q/a
+      const valueSplit = [value.substr(0, value.length - 1), value.slice(-1)]
+
+      // handles nulls
+      if (values[valueSplit[0]]) {
+        // add the value based on the question number and q/a
+        return values[valueSplit[0]][valueSplit[1]]
       }
+
+      return value
     }
-    grid.push(<Triangle row={tri.row} col={tri.col} values={valueArray} config={appConfig.triangle} />)
-  }
+
+    return null;
+  })
 
   return (
-    <g>
-      {grid}
+    <g id={id}>
+      {grid && grid.map((location, index) => {
```

Comment:
This is a conditional render, in that it will check if any value exists for `grid` first and will only render if it does. Again, using an array map to return an array, this time of `Triangle` react components.